### PR TITLE
Wire up DROP MEASUREMENT

### LINKED ISSUE #1695: Wire up DROP MEASUREMENT

### PR DESCRIPTION
**Pull request by corylanou**
Fixes #1695

### CODE REVIEW

**Comment by pauldix** [comment]
LGTM

File: server_test.go [hunk 1]
```diff
@@ -830,6 +830,195 @@ func TestServer_WriteSeries(t *testing.T) {
 	}
 }
 
+// Ensure the server can drop a measurement.
+func TestServer_DropMeasurement(t *testing.T) {
```
Comment (otoolep) [inline server_test.go]: Add a unit test to ensure the the server does the right thing if the Measurement doesn't exist.

File: database.go [hunk 2]
```diff
@@ -1049,6 +1049,35 @@ func (rp *RetentionPolicy) shardGroupByID(shardID uint64) *ShardGroup {
 	return nil
 }
 
+// dropMeasurement will remove a measurement from the index.
+func (db *database) dropMeasurement(name string) error {
```
Comment (otoolep) [inline database.go]: I was expecting this to be different. To drop a measurement, you could just drop an entire bucket in Bolt. I don't think that is what is happening here. Perhaps it doesn't matter.

Comment (corylanou) [inline database.go]: My understanding is the meta store and the shards are different buckets.  Here I need to walk the shards by series that belonged to that measurement to drop them.  In the meta store I can just drop the measurement bucket.  I could of missed something though.

Comment (otoolep) [inline database.go]: Oh, hang on, this is metastore updates? OK, yeah, that will required finer deletions I think (need to check the source). But removing the actual Measurement data? I thought that was just a bucket deletion.

File: database.go [hunk 3]
```diff
@@ -1049,6 +1049,35 @@ func (rp *RetentionPolicy) shardGroupByID(shardID uint64) *ShardGroup {
 	return nil
 }
 
+// dropMeasurement will remove a measurement from the index.
```
Comment (otoolep) [inline database.go]: I'd like this comment to be very precise. The "index" in our system is the in-RAM representation of the what is in the Metastore (which series are in which measurements, etc). The raw data is something else.

So what does this function affect? The in-RAM index, the Metastore on disk, and the actual raw series data?

File: server.go [hunk 4]
```diff
@@ -1606,6 +1606,40 @@ func (s *Server) applyCreateMeasurementsIfNotExists(m *messaging.Message) error
...
+func (s *Server) applyDropMeasurement(m *messaging.Message) error {
+	var c dropMeasurementCommand
+	mustUnmarshalJSON(m.Data, &c)
+
+	database := s.databases[c.Database]
+	if database == nil {
+		return ErrDatabaseNotFound
+	}
+
+	// Remove from metastore.
```
Comment (otoolep) [inline server.go]: This comment implies only the Metastore is affected, but I think there is code in the function below which is also deleting the raw shard data. The comment should make this clear (if I am correct).

**Comment by otoolep** [comment]
Thanks for clarifying comments @corylanou

**Pull request merged**